### DISCUSSION TITLE
Discord OAuth2 Email

### DISCUSSION THREAD

**Discussion by mrvillage** [supabase/supabase#2223] [Questions]
Is there any way to use Discord OAuth2 without taking the user's email? _I'M SO EXCITED FOR DISCORD AUTH, just don't need emails :)_

**Comment by burggraf** [comment]
I don't think so.  It appears that Discord sends email address by default when you authenticate.  I haven't found any settings for this on their developer site.

**Reply by mrvillage** [reply]
Hmm, I implemented my own thing for supabase a while ago (super janky workaround that my computer deleted sadly) and when it wasn't sending me the email address when going to the /user/@me endpoint, but would when I did include the scope. Maybe we did something different, I'm honestly not sure (I don't have much experience with OAuth2, just trying to find a way so I don't need to take people's emails for my service when I have no need of them).

**Reply by burggraf** [reply]
This is certainly possible with some providers, but it varies by provider.

**Reply by mrvillage** [reply]
Huh, I just tested it and it does return an email when you go to /users/@me...thanks for your help! I could've sworn it didn't when I tested it previously and it didn't return the email...

**Reply by burggraf** [reply]
Thanks for the confirmation.

**Reply by mrvillage** [reply]
Upon inquiring with Discord, it turns out that scopes are additive, so if you've already authorized the app with email and don't revoke access then it'll return the email every time you go through the OAuth2 process afterward.

**Reply by burggraf** [reply]
How do you revoke access from their dashboard?

**Reply by mrvillage** [reply]
As far as I'm aware it's on a user basis, so in your Discord account's settings in the client you can go to User Settings -> Authorized Apps then click Deauthorize on the app and the OAuth2 process will be started fresh for that user.

**Reply by prathmesh-mhatre98** [reply]
@mrvillage @burggraf i am facing an email issue with discord OAuth2 when the discord user is signed in via his phone number, supabase takes email in scope by default. Is there any way to remove email from scope ?
